Bind on unix socket: Permission denied

I'm building an image based on this one using Buildah and running it with Podman. The unix socket, for some reason, is not being created with the ownership and permissions I set when building it, which causes MariaDB to fail to start.

All commands are being executed as root.

## To reproduce
1. First, I copy the client-server configuration.
```sh
container=$(buildah from docker.io/library/mariadb:10.5)

buildah copy $container ./client-server.cnf /etc/mysql/conf.d/
```

The `client-server.cnf` file has the following content:
```ini
[client-server]
socket = /var/run/mariadb/mariadb.sock
```

2. Next, I create the socket directory and set ownership and permissions.
```sh
buildah run $container -- mkdir /var/run/mariadb
buildah run $container -- chown -R mysql:user /var/run/mariadb
buildah run $container -- chmod -R 770 /var/run/mariadb
```

"user" is a user and group I create just before running the commands above. Yes, I have tested that the user and group are created successfully.

3. With the image build finished (unmount + commit), I create the pod, a volume for the database files and another volume for the unix socket (to share it with another container later).

This is the command I use to run it:
```sh
podman run -d --pod example --name mariadb --volume example_database:/var/lib/mysql/ --volume example_socket_mariadb:/var/run/mariadb/ localhost/example-mariadb
```

And the result is this:
```
2021-04-02  2:15:41 0 [ERROR] Can't start server : Bind on unix socket: Permission denied
2021-04-02  2:15:41 0 [ERROR] Do you already have another mysqld server running on socket: /var/run/mariadb/mariadb.sock ?
2021-04-02  2:15:41 0 [ERROR] Aborting
2021-04-02 02:16:12+00:00 [ERROR] [Entrypoint]: Unable to start server.
```

## My workaround
- The `example_socket_mariadb` volume will later be mounted into the container it is meant to be shared with. If, after a failed start, I go in and manually fix the permissions of the `/var/run/mariadb` directory from within another running container, the MariaDB container will start successfully after being restarted. The problem here is that the permissions are not being set during the build process for some reason; or maybe they are, but something else is changing them during runtime.

## Things I've tried
- Ran the whole build process from the shell, one step at a time, to pinpoint a failure at some point, but I found none. All commands executed correctly and the permissions of `/var/run/mariadb` were set as expected. I checked that by running:
```
buildah run $container -- ls -al /var/run/mariadb
```

However, when I run the resulting image with podman, the socket directory is owned by `root:root` instead of `mysql:user`.

Before migrating to podman, I used docker. The Dockerfile used to build this image still works, and it has the exact same instructions (in Dockerfile syntax, of course). Maybe this is something related to podman; or maybe it's something I missed; or maybe a problem with the image? I appreciate your help to figure this out.

--- a/10.2/docker-entrypoint.sh
+++ b/10.2/docker-entrypoint.sh
@@ -163,6 +163,8 @@ docker_create_db_directories() {
 	if [ "$user" = "0" ]; then
 		# this will cause less disk access than `chown -R`
 		find "$DATADIR" \! -user mysql -exec chown mysql '{}' +
+		# See https://github.com/MariaDB/mariadb-docker/issues/363
+		find "${SOCKET%/*}" -maxdepth 0 \! -user mysql -exec chown mysql '{}' \;
 	fi
 }
 

--- a/10.3/docker-entrypoint.sh
+++ b/10.3/docker-entrypoint.sh
@@ -163,6 +163,8 @@ docker_create_db_directories() {
 	if [ "$user" = "0" ]; then
 		# this will cause less disk access than `chown -R`
 		find "$DATADIR" \! -user mysql -exec chown mysql '{}' +
+		# See https://github.com/MariaDB/mariadb-docker/issues/363
+		find "${SOCKET%/*}" -maxdepth 0 \! -user mysql -exec chown mysql '{}' \;
 	fi
 }
 

--- a/10.4/docker-entrypoint.sh
+++ b/10.4/docker-entrypoint.sh
@@ -163,6 +163,8 @@ docker_create_db_directories() {
 	if [ "$user" = "0" ]; then
 		# this will cause less disk access than `chown -R`
 		find "$DATADIR" \! -user mysql -exec chown mysql '{}' +
+		# See https://github.com/MariaDB/mariadb-docker/issues/363
+		find "${SOCKET%/*}" -maxdepth 0 \! -user mysql -exec chown mysql '{}' \;
 	fi
 }
 

--- a/10.5/docker-entrypoint.sh
+++ b/10.5/docker-entrypoint.sh
@@ -163,6 +163,8 @@ docker_create_db_directories() {
 	if [ "$user" = "0" ]; then
 		# this will cause less disk access than `chown -R`
 		find "$DATADIR" \! -user mysql -exec chown mysql '{}' +
+		# See https://github.com/MariaDB/mariadb-docker/issues/363
+		find "${SOCKET%/*}" -maxdepth 0 \! -user mysql -exec chown mysql '{}' \;
 	fi
 }
 

--- a/10.6/docker-entrypoint.sh
+++ b/10.6/docker-entrypoint.sh
@@ -163,6 +163,8 @@ docker_create_db_directories() {
 	if [ "$user" = "0" ]; then
 		# this will cause less disk access than `chown -R`
 		find "$DATADIR" \! -user mysql -exec chown mysql '{}' +
+		# See https://github.com/MariaDB/mariadb-docker/issues/363
+		find "${SOCKET%/*}" -maxdepth 0 \! -user mysql -exec chown mysql '{}' \;
 	fi
 }
 

--- a/docker-entrypoint.sh
+++ b/docker-entrypoint.sh
@@ -163,6 +163,8 @@ docker_create_db_directories() {
 	if [ "$user" = "0" ]; then
 		# this will cause less disk access than `chown -R`
 		find "$DATADIR" \! -user mysql -exec chown mysql '{}' +
+		# See https://github.com/MariaDB/mariadb-docker/issues/363
+		find "${SOCKET%/*}" -maxdepth 0 \! -user mysql -exec chown mysql '{}' \;
 	fi
 }
 

